discretize/CylMesh.py

```diff
@@ -146,72 +328,715 @@ def vectorCCx(self):
+    # Grids
+    ####################################################
+
+    @property
+    def _gridNFull(self):
+        """
+        Full Nodal grid (including hanging nodes)
+        """
+        return ndgrid([
+            self.vectorNx, self._vectorNyFull, self.vectorNz
+        ])
+
+    @property
+    def gridN(self):
+        """
+        Nodal grid in cylindrical coordinates :math:`(r, \theta, z)`.
+        Nodes do not exist in a cylindrically symmetric mesh.
+
+        :rtype: numpy.ndarray
+        :return: grid locations of nodes
+        """
+        if self.isSymmetric:
+            self._gridN = self._gridNFull
+        if getattr(self, '_gridN', None) is None:
+            self._gridN = (
+                self._deflationMatrix('N', withHanging=False) *
+                self._gridNFull
+            )
+        return self._gridN
+
+    @property
+    def gridFx(self):
+        """
+        Grid of x-faces (radial-faces) in cylindrical coordinates
+        :math:`(r, \theta, z)`.
+
+        :rtype: numpy.ndarray
+        :return: grid locations of radial faces
+        """
+        if getattr(self, '_gridFx', None) is None:
+            if self.isSymmetric is True:
+                return super(CylMesh, self).gridFx
+            else:
+                self._gridFx = (
+                    self._deflationMatrix('Fx', withHanging=False) *
+                    ndgrid([
+                        self.vectorNx, self.vectorCCy, self.vectorCCz
+                    ])
+                )
+        return self._gridFx
+
+    @property
+    def gridEy(self):
+        """
+        Grid of y-faces (azimuthal-faces) in cylindrical coordinates
+        :math:`(r, \theta, z)`.
+
+        :rtype: numpy.ndarray
+        :return: grid locations of azimuthal faces
+        """
+        if getattr(self, '_gridEy', None) is None:
+            if self.isSymmetric is True:
+                return super(CylMesh, self).gridEy
+            else:
+                self._gridEy = (
+                    self._deflationMatrix('Ey', withHanging=False) *
+                    ndgrid([
+                        self.vectorNx, self.vectorCCy, self.vectorNz
+                    ])
+                )
+        return self._gridEy
+
+    @property
+    def _gridEzFull(self):
+        """
+        Full z-edge grid (including hanging edges)
+        """
+        return ndgrid([
+            self.vectorNx, self._vectorNyFull, self.vectorCCz
+        ])
+
+    @property
+    def gridEz(self):
+        """
+        Grid of z-faces (vertical-faces) in cylindrical coordinates
+        :math:`(r, \theta, z)`.
+
+        :rtype: numpy.ndarray
+        :return: grid locations of radial faces
+        """
+        if getattr(self, '_gridEz', None) is None:
+            if self.isSymmetric is True:
+                self._gridEz = None
+            else:
+                self._gridEz = (
+                    self._deflationMatrix('Ez', withHanging=False) *
```

Comment:
This is just a delete the rows operation?

Reply:
yep, good call, that is much simpler